lib/web/app/auth.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/httplib"
+)
+
+// handleAuth handles authentication for an app
+// When a `POST` request comes in from a trusted proxy address, it'll set the value from the
+// `X-Cookie-Value` header to the `__Host-grv_app_session` cookie.
+func (h *Handler) handleAuth(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+	httplib.SetNoCacheHeaders(w.Header())
+
+	cookieValue := r.Header.Get("X-Cookie-Value")
+	if cookieValue == "" {
+		return trace.BadParameter("X-Cookie-Value header missing")
```

Comment:
Should we log that X-Cookie-Value header is missing and return generic "access denied" error here instead, like we do below?